packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -6581,6 +6581,7 @@ describe('useQuery', () => {
       defaultOptions: { hydrate: { queries: { retry: 1, retryDelay: 10 } } },
     })
     hydrate(clientQueryClient, dehydrated)
+    await sleep(400)
```

Comment:
This is causing the fetch to fail _before_ rendering, which means the client can immediately retry on rendering, putting the query in a fetching state, so the redacted will never get printed to the DOM.

If you instead inspect the queryCache directly, it's there.
If you instead put the await between the two `waitFor` below, the test succeeds again.

So while this might have surfaced an unhandled error in the past, the actual test fail seems to be unrelated.